src/tools/miri/tests/fail/dangling_pointers/dangling_pointer_deref_underscore.rs

```diff
@@ -0,0 +1,11 @@
+// Make sure we find these even with many checks disabled.
+//@compile-flags: -Zmiri-disable-alignment-check -Zmiri-disable-stacked-borrows -Zmiri-disable-validation
+
+fn main() {
+    let p = {
+        let b = Box::new(42);
+        &*b as *const i32
+    };
+    let _ = unsafe { *p }; //~ ERROR: dereferenced after this allocation got freed
+    panic!("this should never print");
+}
```

Comment:
I guess this is also based on https://github.com/rust-lang/rust/pull/104844, or where does this diff come from?  (#104844 is not listed in the PR description.)

Reply:
Yes. Added it to the description.

Reply:
Actually, it doesn't need to be. I'll remove that commit in a few.